ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcInputFormat.java

```diff
@@ -730,7 +729,10 @@ public boolean validateInput(FileSystem fs, HiveConf conf,
           ? AcidOperationalProperties.parseString(txnProperties) : null;
 
       String value = conf.get(ValidWriteIdList.VALID_WRITEIDS_KEY);
-      writeIdList = value == null ? new ValidReaderWriteIdList() : new ValidReaderWriteIdList(value);
+      writeIdList = new ValidReaderWriteIdList(value);
+
+      value = conf.get(ValidTxnList.VALID_TXNS_KEY);
+      validTxnList = new ValidReadTxnList(value);
```

Comment:
Will this help, if we have multiple partitions with multiple files to parse the TxnList only once?